src/core/audio/SNDFileCpp.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Xournal++
+ *
+ * Helper functions for libsndfile
+ * platform dependent code to open std::filesystem::paths
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#ifdef _WIN32
+#include <windows.h>
+#define ENABLE_SNDFILE_WINDOWS_PROTOTYPES 1
+#endif
+
+#include <memory>
+
+#include <sndfile.h>
+
+#include "filesystem.h"
+
+namespace xoj {
+
+struct SNDFILE_Deleter {
+    void operator()(SNDFILE* file) { sf_close(file); }
+};
+
+using SNDFileGuard = std::unique_ptr<SNDFILE, SNDFILE_Deleter>;
+
+
+inline SNDFILE* sf_open_pp(std::string path, int mode, SF_INFO* info) { return sf_open(path.c_str(), mode, info); }
+#if (defined(ENABLE_SNDFILE_WINDOWS_PROTOTYPES) && ENABLE_SNDFILE_WINDOWS_PROTOTYPES)
+inline SNDFILE* sf_open_pp(std::wstring path, int mode, SF_INFO* info) {
+    return sf_wchar_open(path.c_str(), mode, info);
+}
+#endif
```

Comment:
Is there a benefit to taking a string instead of a path and keeping the narrow string overload around?

Otherwise a copy could be avoided (`fs::path::c_str` is equivalient to `.native().c_str()` and `fs::path::native` returns a `const string_type&`):

```suggestion
#ifdef _WIN32
#if (defined(ENABLE_SNDFILE_WINDOWS_PROTOTYPES) && ENABLE_SNDFILE_WINDOWS_PROTOTYPES)
inline SNDFILE* sf_open_pp(fs::path const& path, int mode, SF_INFO* info) {
    return sf_wchar_open(path.c_str(), mode, info);
}
#else // ENABLE_SNDFILE_WINDOWS_PROTOTYPES
#error Inconsistent path char types
#endif // ENABLE_SNDFILE_WINDOWS_PROTOTYPES
#else // _WIN32
inline SNDFILE* sf_open_pp(fs::path const& path, int mode, SF_INFO* info) { return sf_open(path.c_str(), mode, info); }
#endif // _WIN32
```

Reply:
Thanks for the catch, but the functions should take a `StringType const&` with that, automatic overload resolution will find the correct function automatically.